polaris/polaris/deposit/utils.py

```diff
@@ -0,0 +1,139 @@
+import logging
+
+from django.utils.timezone import now
+from stellar_sdk.transaction_builder import TransactionBuilder
+from stellar_sdk.exceptions import BaseHorizonError
+from stellar_sdk.xdr.StellarXDR_type import TransactionResult
+
+from polaris import settings
+from polaris.models import Transaction
+
+
+logger = logging.getLogger(__name__)
+TRUSTLINE_FAILURE_XDR = "AAAAAAAAAGT/////AAAAAQAAAAAAAAAB////+gAAAAA="
+SUCCESS_XDR = "AAAAAAAAAGQAAAAAAAAAAQAAAAAAAAABAAAAAAAAAAA="
+
+
+def create_stellar_deposit(transaction_id: str) -> bool:
+    """
+    Create and submit the Stellar transaction for the deposit.
+
+    :returns a boolean indicating whether or not the deposit was successfully
+        completed. One reason a transaction may not be completed is if a
+        trustline must be established. The transaction's status will be set as
+        ``pending_stellar`` and the status_message will be populated
+        with a description of the problem encountered.
+    :raises ValueError: the transaction has an unexpected status
+    """
+    transaction = Transaction.objects.get(id=transaction_id)
+
+    # We check the Transaction status to avoid double submission of a Stellar
+    # transaction. The Transaction can be either `pending_anchor` if the task
+    # is called from `poll_pending_deposits()` or `pending_trust` if called
+    # from the `check_trustlines()`.
+    if transaction.status not in [
+        Transaction.STATUS.pending_anchor,
+        Transaction.STATUS.pending_trust,
+    ]:
+        raise ValueError(
+            f"unexpected transaction status {transaction.status} for "
+            "create_stellar_deposit",
+        )
+    transaction.status = Transaction.STATUS.pending_stellar
+    transaction.save()
+
+    # We can assume transaction has valid stellar_account, amount_in, and asset
+    # because this task is only called after those parameters are validated.
+    stellar_account = transaction.stellar_account
+    payment_amount = round(transaction.amount_in - transaction.amount_fee, 7)
+    asset = transaction.asset.code
+
+    # If the given Stellar account does not exist, create
+    # the account with at least enough XLM for the minimum
+    # reserve and a trust line (recommended 2.01 XLM), update
+    # the transaction in our internal database, and return.
+
+    server = settings.HORIZON_SERVER
+    starting_balance = settings.ACCOUNT_STARTING_BALANCE
+    server_account = server.load_account(settings.STELLAR_DISTRIBUTION_ACCOUNT_ADDRESS)
+    base_fee = server.fetch_base_fee()
+    builder = TransactionBuilder(
+        source_account=server_account,
+        network_passphrase=settings.STELLAR_NETWORK_PASSPHRASE,
+        base_fee=base_fee
+    )
+    try:
+        server.load_account(stellar_account)
+    except BaseHorizonError as address_exc:
+        # 404 code corresponds to Resource Missing.
+        if address_exc.status != 404:
+            transaction.status_message = (
```

Comment:
do we need to update the `transaction.status` to `error` here as well as `status_message`

Reply:
That is what I had originally had implemented, but I realized that was technically the wrong status to use. See my comment explaining my thought process on `error` vs. `pending_stellar`.